lib/sensors/accelerometer.cpp

```diff
@@ -46,6 +30,28 @@ Accelerometer::~Accelerometer()
 {
 }
 
+std::optional<core::Result> Accelerometer::configure()
+{
+  // check we are communicating with the correct sensor
+  const auto device_id = i2c_->readByte(device_address_, kDeviceIdAddress);
+  if (!device_id) {
+    logger_.log(core::LogLevel::kFatal, "Failed to read the accelerometer device ID");
+    return std::nullopt;
+  }
+  if (*device_id != kExpectedDeviceIdValue) {
+    logger_.log(core::LogLevel::kFatal, "Failure, mismatched device ID for accelerometer");
+    return std::nullopt;
+  }
+  const auto ctrl1_result = i2c_->writeByteToRegister(device_address_, kCtrl1Address, kCtrl1Value);
```

Comment:
Why is this `auto` when we directly check it against failure state 2 lines later? Especially since we wouldn't fail if the return type got changed for whatever reason but would still compile?